File: 프로그래머스/lv1_2016년.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int a, int b) {
    string answer = "";
    string day[7] = {"SUN","MON","TUE","WED","THU","FRI","SAT"};
    int dd = 5;

    if( a != 1 ){
        for(int i=1; i<a; i++){
            if( i == 2 ){
            dd += 29;
            }
            else if( i==1 || i==3 || i==5 || i==7 || i==8 || i==10 || i==12 ){
                dd += 31;
            }
            else {
                dd += 30;
            }
        }
    }
    dd += b-1;
    dd %= 7;
    answer = day[dd];

    return answer;
}

int main(void){
    int a = 2;
    int b = 29;
    string ret = "";

    ret = solution(a,b);

    printf("%s", ret.c_str());
}

File: 프로그래머스/lv1_2016년.py
def solution(a, b):
    answer = ''

    day = ["THU","FRI","SAT","SUN","MON","TUE","WED"]

    month_a = [1,2,3,4,5,6,7,8,9,10,11,12]
    month_b = [31,29,31,30,31,30,31,31,30,31,30,31]

    for a in range(a-1, 0, -1):
        if a in month_a:
            b += month_b[ month_a.index(a) ]

    return day[b % 7]


if __name__ == "__main__":

    a = 2
    b = 1
    ret = solution(a,b)

    print( ret )